iso15118/secc/controller/ev_data.py

```diff
@@ -139,36 +124,398 @@ def __init__(
+    ) -> None:
+        params: Union[
+            ScheduledDCChargeLoopReqParams,
+            DynamicDCChargeLoopReqParams,
+            BPTScheduledDCChargeLoopReqParams,
+            BPTDynamicDCChargeLoopReqParams,
+        ] = None
+        self.present_voltage = dc_charge_loop_req.ev_present_voltage.get_decimal_value()
+        
+
+        if dc_charge_loop_req.display_parameters:
+            _update_display_parameters(dc_charge_loop_req.display_parameters)
+
+        dc_limits = self.session_limits.dc_limits
+        if selected_energy_service.service == ServiceV20.DC:
+            if control_mode == ControlMode.SCHEDULED:
+                params: ScheduledDCChargeLoopReqParams = dc_charge_loop_req.scheduled_params
+                # Target Current and Voltage only in Scheduled
+                self.target_current = params.ev_target_current.get_decimal_value()
+                self.target_voltage = params.ev_target_voltage.get_decimal_value()
+                _update_common_limits(dc_limits, params)
+            elif control_mode == ControlMode.DYNAMIC:
+                params: DynamicDCChargeLoopReqParams = dc_charge_loop_req.dynamic_params
+                # Departure time only in Dynamic
+                if params.departure_time:
+                    self.departure_time = params.departure_time
+                _update_common_limits(dc_limits, params)
+        elif selected_energy_service.service == ServiceV20.DC_BPT:
+            if control_mode == ControlMode.SCHEDULED:
+                params = dc_charge_loop_req.bpt_scheduled_params
+                # Target Current and Voltage only in Scheduled
+                self.target_current = params.ev_target_current.get_decimal_value()
+                self.target_voltage = params.ev_target_voltage.get_decimal_value()
+                _update_common_bpt_limits(dc_limits, params)
+            elif control_mode == ControlMode.DYNAMIC:
+                params = dc_charge_loop_req.bpt_dynamic_params
+                # Departure time only in Dynamic
+                if params.departure_time:
+                    self.departure_time = params.departure_time
+                # V2X Limits only in Dynamic BPT
+                if params.ev_max_v2x_energy_request:
+                    self.max_v2x_energy_request = params.ev_max_v2x_energy_request.get_decimal_value() # noqa: E501
+                if params.ev_min_v2x_energy_request:
+                    self.min_v2x_energy_request = params.ev_min_v2x_energy_request.get_decimal_value() # noqa: E501
+                _update_common_bpt_limits(dc_limits, params) 
+        else:
+            return
+
+        def _update_common_limits(
+            dc_limits: EVDCCLLimits,
+            params: Union[ScheduledDCChargeLoopReqParams,
+                            DynamicDCChargeLoopReqParams,
+                            BPTScheduledDCChargeLoopReqParams,
+                            BPTDynamicDCChargeLoopReqParams]
+        ):
```

Comment:
Would it be easier to read if it is left as member method instead of nested method?